demisto_sdk/commands/common/hook_validations/reputation.py

```diff
@@ -1,19 +1,67 @@
+from distutils.version import LooseVersion
 from demisto_sdk.commands.common.constants import Errors
 from demisto_sdk.commands.common.hook_validations.base_validator import BaseValidator
 from demisto_sdk.commands.common.tools import print_error
 
 
 class ReputationValidator(BaseValidator):
+    """ReputationValidator is designed to validate the correctness of the file structure we enter to content repo.
+    """
+
+    def is_valid_file(self, validate_rn=True):
+        """Check whether the reputation file is valid or not
+        """
+        return all(
+            [
+                self.is_valid_version(),
+                self.is_valid_expiration(),
+                self.is_id_equals_name()
+            ]
+        )
+
     def is_valid_version(self):
         # type: () -> bool
         """Validate that the reputations file as version of -1."""
         is_valid = True
-        reputations = self.current_file.get('reputations', [])
-        for reputation in reputations:
-            internal_version = reputation.get('version')
-            if internal_version != self.DEFAULT_VERSION:
-                object_id = reputation.get('id')
-                print_error(Errors.wrong_version_reputations(self.file_path, object_id, self.DEFAULT_VERSION))
+
+        internal_version = self.current_file.get('version')
+        if internal_version != self.DEFAULT_VERSION:
+            object_id = self.current_file.get('id')
+            print_error(Errors.wrong_version_reputations(self.file_path, object_id, self.DEFAULT_VERSION))
+            is_valid = False
+        return is_valid
+
+    def is_valid_expiration(self):
+        # type: () -> bool
+        """Validate that the expiration field of a 5.5 reputation file is numeric."""
+        error_msg = None
+        is_valid = True
+
+        from_version = self.current_file.get("fromVersion", "0.0.0")
+        if LooseVersion(from_version) >= LooseVersion("5.5.0"):
+            expiration = self.current_file.get('expiration', "")
+            if not isinstance(expiration, int) or expiration < 0:
+                error_msg = f'{self.file_path}: expiration field should have a numeric value.'
+                is_valid = False
+
+        if error_msg:
+            print_error(error_msg)
+        return is_valid
+
+    def is_id_equals_name(self):
+        # type: () -> bool
+        """Validate that the expiration field of a 5.5 reputation file is numeric."""
+        error_msg = None
+        is_valid = True
+
+        from_version = self.current_file.get("fromVersion", "0.0.0")
+        if LooseVersion(from_version) >= LooseVersion("5.5.0"):
```

Comment:
why is it matter what version we are on? why not always enforcing it for new types?